### ISSUE TITLE
[INTEGRATION] Airflow BigQuery Operator enhancement to support routine call.

### ISSUE THREAD

**Issue by fm100** [OpenLineage/OpenLineage#2656]
### What is the target system?

Airflow, BigQuery

### What kind of integration is this?

- [X] Produces OpenLineage metadata
- [ ] Consumes OpenLineage metadata
- [ ] Something else

### How should this integration be implemented?

The current implementation of Airflow BigQueryOperator integration is using the return from the BigQuery API call, particularly `destinationTable` and `referencedTables`. However, BQ API returns nothing when the query is simply a routine call. We might fallback to SQL parsing options if it is a routine. I'll be appreciated for any other idea suggestion!

### Where should this integration be implemented?

- [ ] In the target system
- [X] In the OpenLineage repo
- [ ] Somewhere else

### Do you plan to make this contribution yourself?

- [X] I am interested in doing this work

**Comment by kacpermuda** [comment]
Good idea 😄  Would like to contribute that? I'll be happy to help, should you have any questions.

**Comment by mobuchowski** [comment]
>However, BQ API returns nothing when the query is simply a routine call. We might fallback to SQL parsing options if it is a routine.
 
 It also happens when query is cached.

**Comment by kacpermuda** [comment]
As we'll be removing airflow integration from this repo shortly (#4098) and shift our focus to [OpenLineage Airflow provider](https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/index.html) I'll close this issue. Feel free to reopen in Airflow repo if needed as OpenLineage provider feature request.
